client/src/common/components/Dropdown.jsx

```diff
@@ -15,15 +16,27 @@ class Dropdown extends PureComponent {
     this.dropdown = React.createRef();
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps) {
     const { isVisible } = this.state;
+    const {
+      location: { pathname }
+    } = this.props;
+    const {
+      location: { pathname: prevPathname }
+    } = prevProps;
 
     if (isVisible) {
       document.addEventListener('keydown', this.handleEscapePress);
       document.addEventListener('click', this.handleClickOutside);
+      document.addEventListener('touchend', this.handleClickOutside);
     } else {
       document.removeEventListener('keydown', this.handleEscapePress);
       document.removeEventListener('click', this.handleClickOutside);
+      document.removeEventListener('touchend', this.handleClickOutside);
+    }
+
+    if (pathname !== prevPathname) {
```

Comment:
Please include `isVisible` check to not spam the state changes, i.e. `isVisible && parhname !== previousPathname`.